src/read.c

```diff
@@ -3252,18 +3668,31 @@ static avifResult avifParse(avifDecoder * decoder)
                           AVIF_RESULT_BMFF_PARSE_FAILED);
             moovSeen = AVIF_TRUE;
         }
+#if defined(AVIF_ENABLE_EXPERIMENTAL_AVIR)
+        else if (!memcmp(header.type, "coni", 4)) {
+            AVIF_CHECKERR(!metaSeen && !moovSeen, AVIF_RESULT_BMFF_PARSE_FAILED);
+            AVIF_CHECKRES(avifParseCondensedImageBox(data->meta, boxOffset, boxContents.data, boxContents.size, data->diag));
+            coniSeen = AVIF_TRUE;
+        }
```

Comment:
Nit/optional: Move this else-if earlier, before line 3665. This may make the #if/else-if nesting more regular.

```
            metaSeen = AVIF_TRUE;
#if defined(AVIF_ENABLE_EXPERIMENTAL_AVIR)
        } else if (!memcmp(header.type, "coni", 4)) {
            AVIF_CHECKERR(!metaSeen && !moovSeen, AVIF_RESULT_BMFF_PARSE_FAILED);
            AVIF_CHECKRES(avifParseCondensedImageBox(data->meta, boxOffset, boxContents.data, boxContents.size, data->diag));
            coniSeen = AVIF_TRUE;
#endif
        } else if (!memcmp(header.type, "moov", 4)) {
```

It's a different kind of nesting, so I am not sure if you prefer that.

Reply:
Actually we don't need to move this else-if earlier. We can tweak the #if/else-if nesting at the current location as follows:

```
            moovSeen = AVIF_TRUE;
#if defined(AVIF_ENABLE_EXPERIMENTAL_AVIR)
        } else if (!memcmp(header.type, "coni", 4)) {
            AVIF_CHECKERR(!metaSeen && !moovSeen, AVIF_RESULT_BMFF_PARSE_FAILED);
            AVIF_CHECKRES(avifParseCondensedImageBox(data->meta, boxOffset, boxContents.data, boxContents.size, data->diag));
            coniSeen = AVIF_TRUE;
#endif // AVIF_ENABLE_EXPERIMENTAL_AVIR
        }
```